### PR TITLE
TRITON-1790 Add/resize/remove disk actions to be added to VM#view page

### PR DESCRIPTION
**Pull request by kusor**

### CODE REVIEW

File: www/js/components/pages/vm/disks.js [hunk 1]
```diff
@@ -10,43 +10,133 @@
...
+                return self.collection.length > 0;
             }
         };
-    }
+    },
+
+    onClickAddDisk: function () {
+        var self = this;
+        var view = new DisksFormView({
+            vm: self.vm
```
Comment (twhiteman) [inline www/js/components/pages/vm/disks.js]: Do you need a `model` attribute here? As on line 60 it uses:
```
vm: self.vm,
model: self.model
```

File: www/js/components/pages/vm/disks.js [hunk 2]
```diff
@@ -10,43 +10,133 @@
...
+                this._onJob.bind(this));
+        }
+    },
+    _onJob: function (err, job) {
+        if (err) {
+            console.log('[job error]: ', err);
+            var msg = (err.responseText) ?
+                    JSON.parse(err.responseText).message:
+                    'Error creating job';
+            window.alert(msg);
```
Comment (twhiteman) [inline www/js/components/pages/vm/disks.js]: Should this use a notification instead (as I see some places use notification for job failures):
```
                adminui.vent.trigger('notification', {
                    level:'error',
                    message: msg
                });
```

File: www/js/components/pages/vm/disks-form.hbs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                <div class="form-group">
+                    <div class="controls col-sm-offset-2 col-sm-8">
+                        <div>
+                            <label>SIZE:</label>
+                        </div>
+                        <div>
+
+                        {{#if uuid}}
+                                <input type="hidden" name="pci_slot" value="{{pci_slot}}" />
+                                <input type="text" class="form-control" name="size" value="{{size}}" />
```
Comment (twhiteman) [inline www/js/components/pages/vm/disks-form.hbs]: Indentation.

File: www/js/components/pages/vm/disks-form.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    initialize: function (options) {
+        options = options || {};
+
+        if (options.vm) {
+            this.vm = options.vm;
+        }
+
+        if (!options.model) {
+            this.model = new ViewModel({ size: 0 });
+        }
```
Comment (twhiteman) [inline www/js/components/pages/vm/disks-form.js]: I'm not sure if `this.model` is used? If it is used, then when `options.model` is set, should it be saved to `this.model`?

File: www/js/components/pages/vm/disks-form.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        var dangerous_allow_shrink = this.$('input[name=dangerous_allow_shrink]').is(':checked');
+        var opts = {
+            size: Number(size)
+        };
+        if (pci_slot) {
+            opts.pci_slot = pci_slot;
+            if (dangerous_allow_shrink) {
+                opts.dangerous_allow_shrink = true;
+            }
+            self.vm.resizeDisk(opts, this._onJob.bind(this));
```
Comment (twhiteman) [inline www/js/components/pages/vm/disks-form.js]: Mix of `self` and `this` here is confusing - it looks like `self` is not actually needed?

File: www/js/components/pages/vm/disks-form.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        } else {
+            self.vm.createDisk(opts, this._onJob.bind(this));
```
Comment (twhiteman) [inline www/js/components/pages/vm/disks-form.js]: Mix of `self` and `this` here is confusing - it looks like `self` is not actually needed?

File: www/js/components/pages/vm/disks-form.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        if (err) {
+            console.log('[job error]: ', err);
+            var msg = (err.responseText) ?
+                    JSON.parse(err.responseText).message:
+                    'Error creating job';
+            window.alert(msg);
+            return;
+        }
+        var self = this;
+        self.$el.modal('hide').remove();
```
Comment (twhiteman) [inline www/js/components/pages/vm/disks-form.js]: Mix of `self` and `this` here is confusing.

File: package.json [hunk 8]
```diff
@@ -33,7 +33,7 @@
     "react": "0.13.0",
     "reactify": "1.0.0",
     "restify": "4.3.1",
-    "sdc-clients": "10.2.1",
+    "sdc-clients": "13.0.2",
```
[the remark this answers is not in the document]
Comment (kusor) [reply package.json]: Shouldn't we just get rid of CA as per TRITON-884? Maybe be explicit about this removal?

SAPI is working so far and as far as I can tell, AdminUI hasn't been using DSAPI for a while.

File: www/js/components/pages/vm/disks.js (hunk 1 above)
Comment (kusor) [reply www/js/components/pages/vm/disks.js]: No, b/c on this case the disk is new and we just need the `size` parameter from the form. So, we can just build a model with that single attribute.

File: www/js/components/pages/vm/disks.js (hunk 2 above)
Comment (kusor) [reply www/js/components/pages/vm/disks.js]: Cannot see why not. Changed!

File: www/js/components/pages/vm/disks-form.hbs (hunk 3 above)
Comment (kusor) [reply www/js/components/pages/vm/disks-form.hbs]: Done!

File: www/js/components/pages/vm/disks-form.js (hunk 7 above)
Comment (kusor) [reply www/js/components/pages/vm/disks-form.js]: The only one really needed (`self`) is within the function callback on line 78.

File: www/js/components/pages/vm/disks-form.js (hunk 6 above)
Comment (kusor) [reply www/js/components/pages/vm/disks-form.js]: Yep.

File: www/js/components/pages/vm/disks-form.js (hunk 5 above)
Comment (kusor) [reply www/js/components/pages/vm/disks-form.js]: Changing.

File: www/js/components/pages/vm/disks-form.js (hunk 4 above)
Comment (kusor) [reply www/js/components/pages/vm/disks-form.js]: Actually, the `this.model` is pre-set by React so the `this.model` line here is useless. Removed.

**Pull request merged**